persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/CNASegmentMapper.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+
+<mapper namespace="org.cbioportal.persistence.mybatis.CNASegmentMapper">
+    <cache/>
+
+    <select id="getCNASegmentData" resultType="org.cbioportal.model.CNASegmentData">
+        select s.stable_id as sample, cns.chr as chr, cns.start as start, cns.end as end, cns.num_probes as numProbes, cns.segment_mean as value from sample s, copy_number_seg cns, cancer_study cs 
+        where cs.cancer_study_identifier = #{cancerStudyId} 
+        
+        <if test="sampleIds != null and !sampleIds.isEmpty()">
```

Comment:
Do we really want to allow a user to get the *complete* set of copy number segments for a study?  I guess it depends on how big this return of data could be. If it is many megabytes for a big study, perhaps we want to require a gene list or a sample list?